### PR TITLE
AVX-38356: add test and doc for Edge NEO

### PR DESCRIPTION
**Pull request by shuoliugit**

### CODE REVIEW

File: docs/resources/aviatrix_edge_neo.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+* `site_id` - (Required) Site ID.
+* `device_id` - (Required) Device ID.
+* `gw_size` - (Required) Gateway size. Valid values: "small", "medium", "large", "x-large".
+* `wan_interface_names` - (Required) List of WAN interface names. Default value: ["eth0"].
+* `lan_interface_names` - (Required) List of LAN interface names. Default value: ["eth1"].
+* `management_interface_names` - (Required) List of management interface names. Default value: ["eth2"].
+* `interfaces` - (Required) WAN/LAN/MANAGEMENT interfaces.
+    * `name` - (Required) Interface name.
+    * `type` - (Required) Type. Valid values: WAN, LAN, or MANAGEMENT.
+    * `bandwidth` - (Optional) Bandwidth.
```
Comment (inc1t3Ful) [inline docs/resources/aviatrix_edge_neo.md]: what is bandwidth? What type of values are accepted? Description isnt clear, esp since its not shown in the example code either

File: docs/resources/aviatrix_edge_csp_ha.md [hunk 2]
```diff
@@ -38,11 +38,11 @@ resource "aviatrix_edge_csp_ha" "test" {
...
-* `primary_gw_name` - (Required) Edge CSP name.
+* `primary_gw_name` - (Required) Primary Edge CSP name.
 * `compute_node_uuid` - (Required) Edge CSP compute node UUID.
 
 -> **NOTE:** At least one LAN interface is required.
-* `interfaces` - (Required) WAN/LAN interfaces.
+* `interfaces` - (Required) WAN/LAN/MANAGEMENT interfaces.
   * `name` - (Required) Interface name.
   * `type` - (Required) Type.
   * `bandwidth` - (Optional) Bandwidth.
```
Comment (inc1t3Ful) [inline docs/resources/aviatrix_edge_csp_ha.md]: what is bandwidth? What type of values are accepted? Description isnt clear, esp since its not shown in the example code either

File: docs/resources/aviatrix_edge_neo_ha.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+The following arguments are supported:
+
+### Required
+* `primary_gw_name` - (Required) Primary Edge NEO name.
+* `device_id` - (Required) Device ID.
+* `interfaces` - (Required) WAN/LAN/MANAGEMENT interfaces.
+    * `name` - (Required) Interface name.
+    * `type` - (Required) Type.
+    * `bandwidth` - (Optional) Bandwidth.
```
Comment (inc1t3Ful) [inline docs/resources/aviatrix_edge_neo_ha.md]: what is bandwidth? What type of values are accepted? Description isnt clear, esp since its not shown in the example code either

File: docs/resources/aviatrix_edge_neo_ha.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+subcategory: "Multi-Cloud Transit"
+layout: "aviatrix"
+page_title: "Aviatrix: aviatrix_edge_neo_ha"
+description: |-
+  Creates an Aviatrix Edge NEO HA
+---
+
+# aviatrix_edge_neo_ha
+
+The **aviatrix_edge_neo_ha** resource creates the Aviatrix Edge NEO HA.
```
Comment (inc1t3Ful) [inline docs/resources/aviatrix_edge_neo_ha.md:11]: Are we to assume that in order to use this resource, we need the (primary) **aviatrix_edge_neo** first? If thats the case, we should make a note of that

**Review by inc1t3Ful — CHANGES_REQUESTED** [review]

**Comment by inc1t3Ful** [comment]
Add what the measurement is for bandwidth. What does the integer represent? mbps? kbps? etc. Fix that for all relevant docs and it should be good

**Review by inc1t3Ful — CHANGES_REQUESTED** [review]
Address the above comment regarding bandwidth

**Review by inc1t3Ful — DISMISSED** [review]
LGTM

**Review by zhiqiangjin — APPROVED** [review]
